hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -68,49 +68,27 @@ object InferPType {
         assert(inits.forall(_.nested.isEmpty))
         assert(seqs.forall(_.nested.isEmpty))
         val initArgTypes = inits.map(i => i.value.args.map(_.pType).toArray).transpose
-          .map(ts => getNestedElementPTypes(ts))
+          .map(ts => unifyPTypes(ts))
         val seqArgTypes = seqs.map(i => i.value.args.map(_.pType).toArray).transpose
-          .map(ts => getNestedElementPTypes(ts))
+          .map(ts => unifyPTypes(ts))
         virt.defaultSignature.toPhysical(initArgTypes, seqArgTypes).singletonContainer
     }
   }
 
-  def getNestedElementPTypes(ptypes: Seq[PType]): PType = {
-    assert(ptypes.forall(_.virtualType == ptypes.head.virtualType))
-    getNestedElementPTypesOfSameType(ptypes)
+  def unifyPTypes(pTypes: Seq[PType]): PType = {
```

Comment:
Sure, sounds good. I'm happy to change the name to something without "unify" in the title---my main objection is in calling it "getNestedElementPTypes" because there's a lot of instances where it's not used in a container context. Would "getCompatiblePType" be ok?